CMakeLists.txt

```diff
@@ -0,0 +1,117 @@
+#
+# CMakeLists file for OpenSSL
+# Copyright 2022-2023 The OpenSSL Project Authors. All Rights Reserved.
+#
+# Licensed under the Apache License 2.0 (the "License").  You may not use
+# this file except in compliance with the License.  You can obtain a copy
+# in the file LICENSE in the source distribution or at
+# https://www.openssl.org/source/license.html
+#
+
+cmake_minimum_required(VERSION 3.17)
+
+project(OpenSSL C CXX)
+
+# Configure the OpenSSL project
+if(PERL_EXECUTABLE AND OPENSSL_CONFIGURE_OPTIONS)
+  #set(ENV{BUILDFILE_DEBUG_RULES} 1)
+  #set(ENV{BUILDFILE_DEBUG_DEPENDS} 1)
+  execute_process(
+    COMMAND ${PERL_EXECUTABLE} Configure CMake ${OPENSSL_CONFIGURE_OPTIONS}
+    WORKING_DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}
+    RESULT_VARIABLE OPENSSL_CONFIGURE_RESULT
+  )
+  if(NOT (${OPENSSL_CONFIGURE_RESULT} EQUAL "0"))
+    message(FATAL_ERROR "Failed to configure OpenSSL with error: ${OPENSSL_CONFIGURE_RESULT}")
+  endif()
+endif()
+
+# Include the generated CMake config.
+include(openssl-config.cmake)
```

Comment:
I would suggest to use a different filename here. The pattern `<Base>-config.cmake` typically indicates a CMake config package.